### PR TITLE
Adds top level index page

### LINKED ISSUE #131: Create top level index page
User feedback from Tyrone.

Add top level index page from which all the tables can be assessed and which briefly describes what the numbers in the tables are

### PR DESCRIPTION
**Pull request by wathen**
#### Description of Work

Fixes #131 

Creates a top level index page from a template

#### Testing Instructions

1. Check index page is clear and understandable.
2. Check the buttons work

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: fitbenchmarking/HTML_templates/index_page.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <title>{{ group_name }} results</title>
+</head>
+<body>
+    <section class="row">
+        <header class="large-12 columns">
+            <h1>Fitbenchmarking results: {{ group_name }}</h1>
+            <hr>
+        </header>
+    </section>
+    FitBenchmarking is a tool for comparing different minimizers/fitting software based on their accuracy and runtimes.
```
Comment (tyronerees) [inline fitbenchmarking/HTML_templates/index_page.html]: Maybe just different fitting software (i.e., remove minimizers)?

File: fitbenchmarking/HTML_templates/index_page.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      The runtime results are calculated using the timeit module in python
+      <h3>
+        <form>
+          <button class="btn default" type="submit" formaction={{ rlink }}>Table</button>
+        </form>
+      </h3>    {% endif %}
+
+    {% if compare %}
+      <header class="large-12 columns">
+          <h2>Combine accuracy and runtime results</h2>
```
Comment (tyronerees) [inline fitbenchmarking/HTML_templates/index_page.html]: Combine -> Combined

Can we make this first?  Personally, I don't look at either of the other tables any more.

File: fitbenchmarking/resproc/visual_pages.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import os
 
 from fitbenchmarking.utils.logging_setup import logger
```
Comment (tyronerees) [inline fitbenchmarking/resproc/visual_pages.py]: flake8 complains that logger is imported and not used (not your fault, but might as well fix it here)

File: fitbenchmarking/results_output.py [hunk 4]
```diff
@@ -5,14 +5,17 @@
...
+from jinja2 import Environment, FileSystemLoader
 import logging
 import os
 import pandas as pd
 import pypandoc
+import webbrowser
 
 from fitbenchmarking.resproc import visual_pages
 from fitbenchmarking.utils import create_dirs
 from fitbenchmarking.utils.logging_setup import logger
```
Comment (tyronerees) [inline fitbenchmarking/results_output.py]: flake8 again complains about logger not being used

File: fitbenchmarking/results_output.py [hunk 5]
```diff
@@ -200,18 +203,61 @@ def colour_highlight(value):
...
         # pypandoc can be installed without pandoc
         try:
             output = pypandoc.convert_file(name + 'html', 'rst')
             with open(name + 'rst', "w") as f:
                 f.write(output)
         except ImportError:
             print('RST tables require Pandoc to be installed')
+
+
+def create_top_level_index(options, table_names,group_name):
```
Comment (tyronerees) [inline fitbenchmarking/results_output.py]: pep8 requires a space after a comma

File: fitbenchmarking/HTML_templates/index_page.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<!DOCTYPE html>
+<html>
+<meta charset="UTF-8">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: I have removed this and added a blank css style sheet which we can use

File: fitbenchmarking/HTML_templates/index_page.html [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Raleway">
+<style>
+body,h1,h2,h3,h4,h5 {ofnt-family: "Raleway", sans-serif}
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: correct with new css style sheet

File: fitbenchmarking/HTML_templates/index_page.html [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+table {
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: correct with new css style sheet

File: fitbenchmarking/HTML_templates/index_page.html [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+}
+</style>
+
+<html>
+<head>
+    <title>{{ group_name }} results</title>
+</head>
+<body>
+    <section class="row">
+        <header class="large-12 columns">
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: Changed large-12 to large

File: fitbenchmarking/HTML_templates/style_sheet.html [hunk 10]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/HTML_templates/style_sheet.html]: correct with new css style sheet

File: fitbenchmarking/resproc/visual_pages.py [hunk 11]
```diff
@@ -81,11 +82,16 @@ def create(prob_results, group_name, results_dir, count):
...
+    html_dir = file_path.rsplit('/', 1)
     root = os.path.dirname(os.path.abspath(__file__))
-    env = Environment(loader=FileSystemLoader(root))
+    env = Environment(loader=FileSystemLoader(html_dir[0]))
 
-    template = env.get_template('results_template.html')
+    template_html = '{}/../HTML_templates/results_template.html'.format(root)
+    style_html = '{}/../HTML_templates/style_sheet.html'.format(root)
     html_link = "{0}.html".format(file_path)
+    combine_files(html_link, style_html, template_html)
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/resproc/visual_pages.py]: Been removed due to new css style sheet

File: fitbenchmarking/results_output.py [hunk 12]
```diff
@@ -200,18 +203,61 @@ def colour_highlight(value):
...
+    :param group_name : name of the problem group
+    :type group_name : str
+    """
+    root = os.path.dirname(os.path.abspath(__file__))
+
+    env = Environment(loader=FileSystemLoader(
+        table_names.values()[0].rsplit('/', 1)[0]))
+
+    template_html = '{}/HTML_templates/index_page.html'.format(root)
+    style_html = '{}/HTML_templates/style_sheet.html'.format(root)
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/results_output.py]: correct with new css style sheet

File: fitbenchmarking/utils/misc.py [hunk 13]
```diff
@@ -35,3 +35,31 @@ def get_problem_files(data_dir):
         logger.info(problem)
 
     return problems
+
+
+def combine_files(output_file, *files):
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/utils/misc.py]: Correct, has been removed

File: fitbenchmarking/HTML_templates/index_page.html (hunk 2 above)
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: Corrected

File: fitbenchmarking/HTML_templates/index_page.html (hunk 1 above)
Comment (wathen) [reply fitbenchmarking/HTML_templates/index_page.html]: Corrected

File: fitbenchmarking/resproc/visual_pages.py (hunk 3 above)
Comment (wathen) [reply fitbenchmarking/resproc/visual_pages.py]: removed logger

File: fitbenchmarking/results_output.py (hunk 4 above)
Comment (wathen) [reply fitbenchmarking/results_output.py]: removed logger

File: fitbenchmarking/results_output.py (hunk 5 above)
Comment (wathen) [reply fitbenchmarking/results_output.py]: corrected

File: fitbenchmarking/HTML_templates/index_page.html (hunk 1 above)
Comment (tyronerees) [reply fitbenchmarking/HTML_templates/index_page.html]: This doesn't seem to have been updated - has it been pushed?

**Pull request merged**